android/hello_sdl_android/src/main/java/com/sdl/hellosdlandroid/SdlReceiver.java

```diff
@@ -1,18 +1,29 @@
 package com.sdl.hellosdlandroid;
 
 import android.app.PendingIntent;
+import android.content.BroadcastReceiver;
 import android.content.Context;
 import android.content.Intent;
+import android.content.IntentFilter;
+import android.hardware.usb.UsbAccessory;
+import android.hardware.usb.UsbManager;
 import android.os.Build;
 
 import com.smartdevicelink.transport.SdlBroadcastReceiver;
 import com.smartdevicelink.transport.SdlRouterService;
 import com.smartdevicelink.transport.TransportConstants;
+import com.smartdevicelink.util.AndroidTools;
 import com.smartdevicelink.util.DebugTool;
 
+import java.lang.ref.WeakReference;
+
 public class SdlReceiver extends SdlBroadcastReceiver {
     private static final String TAG = "SdlBroadcastReceiver";
 
+    private static final String ACTION_USB_PERMISSION = "com.android.example.USB_PERMISSION";
+    private PendingIntent pendingIntent;
+    private Intent cachedIntent;
```

Comment:
These should probably be a little more specific in terms of what they are without having to read into the code more to find out what they do. These are verbose names though so maybe you can think of something better. 

```suggestion
    private PendingIntent pendingIntentToStartService;
    private Intent startSdlServiceIntent;
```